unity-environment/Assets/ML-Agents/Examples/Soccer/Scripts/AgentSoccer.cs

```diff
@@ -143,52 +95,47 @@ public void MoveAgent(float[] act)
         if (agentRole == AgentRole.goalie)
         {
             kickPower = 0f;
-            if (action == 0)
-            {
-                dirToGo = transform.forward * 1f;
-                kickPower = 1f;
-            }
-            else if (action == 1)
-            {
-                dirToGo = transform.forward * -1f;
-            }
-            else if (action == 3)
-            {
-                dirToGo = transform.right * -1f;
-            }
-            else if (action == 2)
+            switch (action)
             {
-                dirToGo = transform.right * 1f;
+                case 0:
+                    dirToGo = transform.forward * 1f;
+                    kickPower = 1f;
+                    break;
+                case 1:
+                    dirToGo = transform.forward * -1f;
+                    break;
+                case 3:
+                    dirToGo = transform.right * -1f;
+                    break;
+                case 2:
+                    dirToGo = transform.right * 1f;
+                    break;
             }
-
         }
         else
         {
             kickPower = 0f;
-            if (action == 0)
-            {
-                dirToGo = transform.forward * 1f;
-                kickPower = 1f;
-            }
-            else if (action == 1)
-            {
-                dirToGo = transform.forward * -1f;
-            }
-            else if (action == 2)
-            {
-                rotateDir = transform.up * 1f;
-            }
-            else if (action == 3)
-            {
-                rotateDir = transform.up * -1f;
-            }
-            else if (action == 4)
-            {
-                dirToGo = transform.right * -1f;
-            }
-            else if (action == 5)
+            switch (action)
             {
-                dirToGo = transform.right * 1f;
+                case 0:
+                    dirToGo = transform.forward * 1f;
+                    kickPower = 1f;
+                    break;
+                case 1:
+                    dirToGo = transform.forward * -1f;
+                    break;
+                case 2:
+                    rotateDir = transform.up * 1f;
+                    break;
+                case 3:
+                    rotateDir = transform.up * -1f;
+                    break;
+                case 4:
+                    dirToGo = transform.right * -1f;
```

Comment:
Can this be smaller than 1 to force the agent to move forward rather than in a crab fashion ?

Reply:
The agent can only "kick" the ball when going forward though.